pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -103,13 +99,18 @@ func (q *tenantQuerier) Close() error {
 // If the label `idLabelName` already exists, its value is overwritten and
 // the previous value is exposed through a new label prefixed with "original_".
 // This behaviour is not implemented recursively.
-func NewMergeQueryable(idLabelName string, callbacks MergeQueryableCallbacks, bypassWithSingleID bool, maxConcurrency int, logger log.Logger) storage.Queryable {
+func NewMergeQueryable(idLabelName string, callbacks MergeQueryableCallbacks, bypassWithSingleID bool, maxConcurrency int, reg prometheus.Registerer, logger log.Logger) storage.Queryable {
 	return &mergeQueryable{
 		logger:             logger,
 		idLabelName:        idLabelName,
 		callbacks:          callbacks,
 		bypassWithSingleID: bypassWithSingleID,
 		maxConcurrency:     maxConcurrency,
+		tenantsQueried: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+			Name:    "cortex_querier_federation_sample_tenants_queried",
```

Comment:
I don't like the name of this metrics but I can't think of a better one. Thoughts?

Reply:
Perhaps to mirror the difference between `ExemplarQueryable` and `Queryable`, this could just be `cortex_querier_federation_tenants_queried`?